src/solve.jl

```diff
@@ -61,6 +61,9 @@ function solve!(problem::InitialValueProblem,
                 op::ContinuousPost=default_operator(problem)
                )
 
+    @assert statedim(problem) == dim(problem.x0) "the state-space dimension should match the " *
+    "dimension of the initial states, but they are of size $(statedim(p)) and $(problem.x0) respectively"
```

Comment:
You check `statedim(problem) == dim(problem.x0)` but the error outpus `$(statedim(p)) and $(problem.x0)`.
A test for an `AssertionError` would find this.

Reply:
Thanks for spotting this, i've fixed it and added a test.